ext/intl/breakiterator/breakiterator_iterators.cpp

```diff
@@ -234,23 +228,23 @@ void IntlIterator_from_BreakIterator_parts(zval *break_iter_zv,
 	object_init_ex(object, IntlPartsIterator_ce_ptr);
 	ii = Z_INTL_ITERATOR_P(object);
 
-	ii->iterator = (zend_object_iterator*)emalloc(sizeof(zoi_break_iter_parts));
+	ii->iterator = static_cast<zend_object_iterator *>(emalloc(sizeof(zoi_break_iter_parts)));
 	zend_iterator_init(ii->iterator);
 
 	ZVAL_COPY(&ii->iterator->data, break_iter_zv);
 	ii->iterator->funcs = &breakiterator_parts_it_funcs;
 	ii->iterator->index = 0;
 
-	((zoi_with_current*)ii->iterator)->destroy_it = _breakiterator_parts_destroy_it;
-	ZVAL_OBJ_COPY(&((zoi_with_current*)ii->iterator)->wrapping_obj, Z_OBJ_P(object));
-	ZVAL_UNDEF(&((zoi_with_current*)ii->iterator)->current);
+	reinterpret_cast<zoi_with_current *>(ii->iterator)->destroy_it = _breakiterator_parts_destroy_it;
```

Comment:
While at it, just assign the result of the reinterpret_cast out to another variable and use that variable instead of repeating the reinterpret_cast a dozen times